app/components/preprint-form-authors.js

```diff
@@ -10,26 +10,158 @@ export default CpPanelBodyComponent.extend(PreprintFormFieldMixin, {
     permissionChanges: {},
     bibliographicChanges: {},
     valid: Ember.computed.alias('newContributorId'),
+    permissionToggle: false,
+    bibliographicToggle: false,
+    removalToggle: false,
+    stillAdmin: Ember.computed('isAdmin', function() {
+        return this.get('isAdmin');
+    }),
+    addState: 'emptyView', // There are 3 view states on left side of Authors panel. Default state just shows search bar.
+    query: null,
+    // Total contributor search results
+    totalSearchResults: Ember.computed('searchResults.[]', function() {
+        let searchResults = this.get('searchResults');
+        if (searchResults && searchResults.links) {
+            return Math.ceil(searchResults.links.meta.total);
+        } else {
+            return;
+        }
+    }),
+    // Total pages of contributor search results
+    pages: Ember.computed('totalSearchResults', function() {
+        let totalSearchResults = this.get('totalSearchResults');
+        return Math.ceil(totalSearchResults / 10);
+    }),
+    // Search results excluding users that are already contributors
+    newSearchResults: Ember.computed('searchResults.[]', 'contributors.[]', 'addState', function() {
+        let searchResults = this.get('searchResults');
+        let contributors = this.get('contributors');
+        let userIds = contributors.map((contrib) => contrib.id.split('-')[1]);
+        return searchResults.filter((result) => !Ember.A(userIds).contains(result.id));
+    }),
     actions: {
-        addContributor(userId, permission, isBibliographic) {
-            this.sendAction('addContributor', userId, permission, isBibliographic);
+        // Adds contributor then redraws view - addition of contributor may change which update/remove contributor requests are permitted
+        addContributor(user) {
+            this.sendAction('addContributor', user.id, 'write', true);
+            this.toggleProperty('permissionToggle');
+            this.toggleProperty('bibliographicToggle');
+            this.toggleProperty('removalToggle');
+        },
+        // Adds unregistered contributor, then clears form and switches back to search view.
+        // Should wait to transition until request has completed.
+        addUnregisteredContributor(fullName, email) {
+            let res = this.attrs.addUnregisteredContributor(fullName, email, 'write', true);
+            var _this = this;
+            res.then(function() {
+                _this.toggleProperty('bibliographicToggle');
+                _this.toggleProperty('permissionToggle');
+                _this.toggleProperty('removalToggle');
+            });
+            this.set('addState', 'searchView');
```

Comment:
You may want to put this stuff inside a single reset function to DRY future cleanup.

Additionally, just as an aside, there is a function called `setProperties` that can set multiple things at once. Not a big issue, but convenient.